File: framework/src/main/java/com/github/mrmitew/skeleton/base/view/InjectableFragment.java
package com.github.mrmitew.skeleton.base.view;

import android.os.Bundle;
import android.support.v4.app.Fragment;

import com.github.mrmitew.skeleton.di.fragment.HasFragmentSubcomponentBuilders;

public abstract class InjectableFragment extends Fragment {
    private static final String TAG = "InjectableFragment";

    public InjectableFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setupFragmentComponent();
    }

    protected void setupFragmentComponent() {
        injectMembers((HasFragmentSubcomponentBuilders) getActivity());
    }

    protected abstract void injectMembers(HasFragmentSubcomponentBuilders
                                                  hasFragmentSubcomponentBuilders);
}

File: framework/src/main/java/com/github/mrmitew/skeleton/base/view/InjectableFragment.kt
package com.github.mrmitew.skeleton.base.view

import android.os.Bundle
import android.support.v4.app.Fragment
import com.github.mrmitew.skeleton.di.fragment.HasFragmentSubComponentBuilders

abstract class InjectableFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setupFragmentComponent()
    }

    protected fun setupFragmentComponent() {
        injectMembers(activity as HasFragmentSubComponentBuilders)
    }

    protected abstract fun injectMembers(hasFragmentSubcomponentBuilders: HasFragmentSubComponentBuilders)


}